ledger/participant-integration-api/src/main/scala/platform/store/dao/events/EventsTableFlatEvents.scala

```diff
@@ -43,30 +44,53 @@ private[events] object EventsTableFlatEvents {
         )
     }
 
-  private val archivedFlatEventParser: RowParser[EventsTable.Entry[Raw.FlatEvent.Archived]] =
-    EventsTable.archivedEventRow map {
-      case eventOffset ~ transactionId ~ nodeIndex ~ eventSequentialId ~ eventId ~ contractId ~ ledgerEffectiveTime ~ templateId ~ commandId ~ workflowId ~ eventWitnesses =>
-        // ArraySeq.unsafeWrapArray is safe here
-        // since we get the Array from parsing and don't let it escape anywhere.
-        EventsTable.Entry(
-          eventOffset = eventOffset,
-          transactionId = transactionId,
-          nodeIndex = nodeIndex,
-          eventSequentialId = eventSequentialId,
-          ledgerEffectiveTime = ledgerEffectiveTime,
-          commandId = commandId.getOrElse(""),
-          workflowId = workflowId.getOrElse(""),
-          event = Raw.FlatEvent.Archived(
-            eventId = eventId,
-            contractId = contractId,
-            templateId = templateId,
-            eventWitnesses = ArraySeq.unsafeWrapArray(eventWitnesses),
-          ),
-        )
+  private val exercisedFlatEventParser: RowParser[EventsTable.Entry[Raw.FlatEvent]] =
```

Comment:
This is a common parser that can parse both `Archived` and non-consuming `Exercised` events. The differentiation is made by checking the value of `exercise_consuming` column from `participant_events` table.